tests/devices/test_default_qutrit.py

```diff
@@ -0,0 +1,570 @@
+        "operation,input,expected_output,par", test_data_two_wires_with_parameters
+    )
+    def test_apply_operation_two_wires_with_parameters(
+        self, qutrit_device_2_wires, tol, operation, input, expected_output, par
+    ):
+        """Tests that applying an operation yields the expected output state for two wire
+        operations that have parameters."""
+
+        qutrit_device_2_wires._state = np.array(input, dtype=qutrit_device_2_wires.C_DTYPE).reshape(
+            (3, 3)
+        )
+        qutrit_device_2_wires.apply([operation(par, wires=[0, 1])])
+
+        assert np.allclose(
+            qutrit_device_2_wires._state.flatten(), np.array(expected_output), atol=tol, rtol=0
+        )
+        assert qutrit_device_2_wires._state.dtype == qutrit_device_2_wires.C_DTYPE
+
+    @pytest.mark.parametrize(
+        "operation,input,expected_output,par", test_data_two_wires_with_parameters_inverse
+    )
+    def test_apply_operation_two_wires_with_parameters_inverse(
+        self, qutrit_device_2_wires, tol, operation, input, expected_output, par
+    ):
+        """Tests that applying an operation yields the expected output state for two wire
+        operations that have parameters."""
+
+        qutrit_device_2_wires._state = np.array(input, dtype=qutrit_device_2_wires.C_DTYPE).reshape(
+            (3, 3)
+        )
+        qutrit_device_2_wires.apply([operation(par, wires=[0, 1]).inv()])
+
+        assert np.allclose(
+            qutrit_device_2_wires._state.flatten(), np.array(expected_output), atol=tol, rtol=0
+        )
+        assert qutrit_device_2_wires._state.dtype == qutrit_device_2_wires.C_DTYPE
+
+    def test_apply_rotations_one_wire(self, qutrit_device_1_wire, tol):
+        """Tests that rotations are applied in correct order after operations"""
+
+        state = [1, 0, 0]
+        qutrit_device_1_wire._state = np.array(state, dtype=qutrit_device_1_wire.C_DTYPE)
+
+        ops = [
+            qml.QutritUnitary(U_shift, wires=0).inv(),
+            qml.QutritUnitary(U_thadamard_01, wires=0),
+        ]
+        rotations = [
+            qml.QutritUnitary(U_thadamard_01, wires=0),
+            qml.QutritUnitary(U_shift, wires=0),
+        ]
+
+        qutrit_device_1_wire.apply(ops, rotations)
+
+        assert np.allclose(qutrit_device_1_wire._state.flatten(), state)
+
+    # TODO: Add tests for state preperation ops after they're implemented
+
+
+class TestDefaultQutritIntegration:
+    """Integration tests for default.qutrit. This test ensures it integrates
+    properly with the PennyLane interface, in particular QNode."""
+
+    def test_defines_correct_capabilities(self):
+        """Test that the device defines the right capabilities"""
+
+        dev = qml.device("default.qutrit", wires=1)
+        cap = dev.capabilities()
+        capabilities = {
+            "model": "qutrit",
+            "supports_finite_shots": True,
+            "supports_tensor_observables": True,
```

Comment:
Just to make sure: this is coming up in a subsequent PR, correct?

Reply:
@mudit2812 can confirm but yes, observables are coming soon!

Reply:
Just finished testing the last of the observables that are being added for qutrits (will come in a later PR)!